src/ntfs/read.rs

```diff
@@ -110,7 +110,13 @@ pub fn manage_fares_v1(collections: &mut Collections, base_path: &path::Path) ->
     let file_od_fares = "od_fares.csv";
     let file_fares = "fares.csv";
 
-    if !base_path.join(file_prices).exists() || !base_path.join(file_od_fares).exists() {
+    if !base_path.join(file_prices).exists()
+        || !base_path.join(file_od_fares).exists()
+        || (!collections.tickets.is_empty()
+            && !collections.ticket_uses.is_empty()
+            && !collections.ticket_prices.is_empty()
+            && !collections.ticket_use_perimeters.is_empty())
```

Comment:
Note that even if `ticket_use_restrictions.txt` is optional, I would still test for its presence.

Or the other option is, as long as `collections.tickets` is empty, we assume there isn't any Fares V2 in the model, no need to do an exhaustive test. This is in theory a correct solution. But I would keep the exhaustive solution like you already did in order to be more defensive in our code.